library/src/main/java/com/ricohapi/mstorage/MediaStorage.java

```diff
@@ -65,292 +63,387 @@ public void onThrowable(Throwable t) {
         });
     }
 
-    public void upload(InputStream inputStream, CompletionHandler<MediaInfo> handler) {
-        try {
-            if (accessToken == null) {
-                throw new RicohAPIException(0, "wrong usage: use the connect method to get an access token.");
+    public void upload(final InputStream inputStream, final CompletionHandler<MediaInfo> handler) {
+        authClient.getAccessToken(new CompletionHandler<AuthResult>(){
+            @Override
+            public void onCompleted(AuthResult result) {
+                try {
+                    if (result.getAccessToken() == null) {
+                        throw new RicohAPIException(0, "wrong usage: use the connect method to get an access token.");
+                    }
+                    RicohAPIRequest request = new RicohAPIRequest(ENDPOINT);
+                    Map<String, String> header = new HashMap<>();
+                    header.put("Authorization", "Bearer " + result.getAccessToken());
+                    header.put("Content-Type", "image/jpeg");
+                    request.upload(header, inputStream);
+
+                    if (request.isSucceeded()) {
+                        MediaInfo mediaInfo = JSON.decode(request.getResponseBody(), MediaInfo.class);
+                        handler.onCompleted(mediaInfo);
+                    } else {
+                        throw new RicohAPIException(request.getResponseCode(), request.getErrorBody());
+                    }
+                } catch (IOException e) {
+                    handler.onThrowable(e);
+                } catch (RicohAPIException e) {
+                    handler.onThrowable(e);
```

Comment:
`RicohAPIException` can be caught because of either wrong usage(such as not using `connect` first) or change in response body. For the latter, I do not think that users would be happy to see the raw response body of refreshing request when they're trying to use `upload`, so showing the caught error itself might not be a good idea. Maybe we can just make an error and put an original message in it like L72.